### PR TITLE
LLVM parser/highlighter added

### PR DESCRIPTION
**Pull request by benwilliamgraham**

### CODE REVIEW

File: lua/nvim-treesitter/parsers.lua [hunk 1]
```diff
@@ -738,6 +738,14 @@ list.hjson = {
   maintainers = { "@winston0410" },
 }
 
+list.llvm = {
+  install_info = {
+    url = "https://github.com/benwilliamgraham/tree-sitter-llvm",
+    files = { "src/parser.c" },
```
Comment (theHamsta) [inline lua/nvim-treesitter/parsers.lua:744]: branch = "main",

Comment (benwilliamgraham) [reply lua/nvim-treesitter/parsers.lua:745]: Good catch!

**Comment by theHamsta** [comment]
Nice!

This statement causes a parser error (on FullDebug)
```
!0 = distinct !DICompileUnit(language: DW_LANG_C99, file: !1, producer: "clang", isOptimized: false, runtimeVersion: 0, emissionKind: FullDebug, enums: !2, retainedTypes: !2, globals: !2, imports: !2)
```

Is it on purpose that the parser doesn't expose any hierarchy on the AST?

**Review by theHamsta — APPROVED** [review]

**Comment by benwilliamgraham** [comment]
> Nice!
> 
> This statement causes a parser error (on FullDebug)
> 
> ```
> !0 = distinct !DICompileUnit(language: DW_LANG_C99, file: !1, producer: "clang", isOptimized: false, runtimeVersion: 0, emissionKind: FullDebug, enums: !2, retainedTypes: !2, globals: !2, imports: !2)
> ```

Nice catch, I'll update that now!

> Is it on purpose that the parser doesn't expose any hierarchy on the AST?
> 
> `ret` could be `@keyword.return`

Yeah, for now I am modeling the parser after LLVM's existing vim highlighter, but in the future I plan on adding more structure to facilitate folding/etc.

File: queries/llvm/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+(number) @number
+(comment) @comment
+(string) @string
+(label) @label
+(keyword) @keyword
+(boolean) @boolean
+(float) @number
+(constant) @identifier
+(identifier) @constant
+(punctuation) @punctuation
```
Comment (theHamsta) [inline queries/llvm/highlights.scm:12]: I think this causes the CI to fail. We only have punctuation.delimiter

Comment (benwilliamgraham) [reply queries/llvm/highlights.scm]: Gotcha, should be fixed now!

**Review by theHamsta — APPROVED** [review]

**Pull request merged**